js/plugins/iq.js

```diff
@@ -236,7 +246,29 @@ var IqPlugin= function (framework)
         this.ooyalaReporter.reportReplay();
         OO.log("IQ: Reported: reportReplay()");
         break;
-
+      //OO.EVENTS.BUFFERING -> OO.Analytics.EVENTS.VIDEO_BUFFERING_STARTED.
+      case OO.Analytics.EVENTS.VIDEO_BUFFERING_STARTED: 
+        eventParams = params[0];
+        eventMetadata = {};
+        eventMetadata.qosEventName = eventName;
+        eventMetadata.position = eventParams.position;
+        OO.log("IQ: Reported: reportCustomEvent() for event: " + eventName + " with args:" + JSON.stringify(eventMetadata));
+        this.ooyalaReporter.reportCustomEvent(eventName, eventMetadata);
+        break;
+      case OO.Analytics.EVENTS.INITIAL_PLAY_STARTING:
+      case OO.Analytics.EVENTS.PLAYBACK_READY:
+      case OO.Analytics.EVENTS.API_ERROR:
+      case OO.Analytics.EVENTS.BITRATE_INITIAL:
+      case OO.Analytics.EVENTS.BITRATE_FIVE_SEC:
+      case OO.Analytics.EVENTS.BITRATE_STABLE:
+      case OO.Analytics.EVENTS.PLAYBACK_START_ERROR:
+      case OO.Analytics.EVENTS.PLAYBACK_MIDSTREAM_ERROR:
+      case OO.Analytics.EVENTS.PLUGIN_LOADED:
+        eventMetadata = params[0];
```

Comment:
Can we add a null check to see if params and params[0] exists?

Reply:
Added